portfolio.py

```diff
@@ -2,13 +2,29 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import ENCODE_NP
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
     def __init__(self, date):
         dict.__init__(self)
         self.date = date
+        self.position_np = None
+
+    def encode_np(self, price_dataset):
```

Comment:
The name should be `encode_to_nparray`, which is more readable.
Besides, can you implement the corresponding decoding method, `decode_from_nparray`?